PowerPointLabs/PowerPointLabs/SyncLab/SyncFormatUtil.cs

```diff
@@ -54,5 +55,180 @@ public static bool IsValidFormatName(string name)
+                    // charts are not yet supported by Synclab
+                    break;
+                case PpPlaceholderType.ppPlaceholderTable:
+                    // tables are not yet supported by Synclab
+                    break;
+                case PpPlaceholderType.ppPlaceholderPicture:
+                case PpPlaceholderType.ppPlaceholderBitmap:
+                    // TODO: images are not yet supported by SyncLab, see PictureFormat for things to copy
+                    // do nothing for now
+                    break;
+                case PpPlaceholderType.ppPlaceholderVerticalObject:
+                case PpPlaceholderType.ppPlaceholderObject:
+                    // already narrowed down the type 
+                    // should only perform actions valid for all placeholder objects here 
+                    // do nothing for now
+                    break;
+                default:
+                    // types not listed above are types that do not make sense to be supported by Synclab
+                    // eg. footer, header, date
+                    break;
+            }
+            
+            if (shapeTemplate == null)
+            {
+                // placeholder type is not supported, no copy made
+                return null;
+            }
+            
+            ApplyFormats(formats, msoPlaceHolder, shapeTemplate);
+            return shapeTemplate;
+        }
+
+        /// <summary>
+        /// Targets only msoPlaceHolderObject & msoPlaceHolderVerticalObject
+        /// Attempt to return a more specific placeholder type, if we can determine it
+        /// </summary>
+        /// <param name="placeHolder"></param>
+        /// <returns>a specific type, or the shape's original type</returns>
+        private static PpPlaceholderType GetSpecificPlaceholderType(Shape placeHolder)
+        {
+            bool isPicture = IsPlaceHolderPicture(placeHolder);
+            bool isBody = IsPlaceHolderBody(placeHolder);
+            if (isPicture)
+            {
+                return PpPlaceholderType.ppPlaceholderPicture;
+            }
+            else if (isBody)
+            {
+                return PpPlaceholderType.ppPlaceholderBody;
+            }
+            else
+            {
+                return placeHolder.PlaceholderFormat.Type;
+            }
+        }
+
+        /// <summary>
+        /// Checks if a placeholder is a Picture
+        /// shape.PlaceHolderFormat.Type is insufficient, sometimes returning the more general "ppPlaceHolderObject".
+        /// </summary>
+        /// <param name="placeHolder"></param>
+        /// <returns></returns>
+        private static bool IsPlaceHolderPicture(Shape placeHolder)
+        {
+            try
+            {
+                // attempt to access PictureFormat properties, an exception will be thrown
+                // if shape is not a Picture.
+                float unused = placeHolder.PictureFormat.CropTop;
```

Comment:
ignoring, we need to run the code to throw an exception.

Reply:
You can just run the code without assigning the variable

Reply:
Not assigning was caught by resharper's linting so I assigned it. I'll swap over so it looks better on SonarCube if that's preferred.

Reply:
Interestingly, not using a variable is seen as a compilation error, [cs0201](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0201). (mentioned by resharper)

I guess the error is not enforced by VS2015 but I'll leave the code as it is, since MS's docs prefers assigning to a variable.